File: Frontend/src/app/models/Expense/expense.ts

import { Base } from "../Base.js";
import { Frequency } from "../Admin/frequency.js";
import { User } from "../Admin/user.js";
import { UserPaymentMethod } from "../Admin/UserPaymentMethod.js";
import { ExpenseCategory } from "../Admin/ExpenseCategory.js";
import { ExpenseType } from "../Admin/ExpenseType.js";

export class Expense extends Base 
{   
    public user? : User;
    public frequency? : Frequency;
    public paid? : boolean;
    public value? : number;
    public paidTo? : string;
    public paidVia? : UserPaymentMethod;
    public description?: string;
    public expenseType?: ExpenseType;
    public expenseCategory?: ExpenseCategory;
    public paidOn? : Date;

}

File: Features/DL/Entities/Expense.cs
namespace Features.DL.Entities
{
    public class Expense: BaseModel
    {

        public virtual User User { get; set; }
        public virtual Frequency Frequency { get; set; }
        public virtual Boolean Paid {  get; set; }

        public virtual Decimal Value { get; set; }  
        public virtual string PaidTo { get; set; }

        public virtual UserPaymentMethod PaidVia { get; set; }

        public virtual ExpenseType ExpenseType { get; set; }

        public virtual ExpenseCategory ExpenseCategory { get; set; }
        public virtual string Description { get; set; }

        public virtual DateTime? PaidOn { get; set; }
    }
}
